### PR TITLE
Adding authToken validation in bulkUpload

### PR DESCRIPTION
**Pull request by Sahil-tarento**

### CODE REVIEW

File: src/main/java/org/sunbird/user/registration/model/UserRegistration.java [hunk 1]
```diff
@@ -101,6 +102,14 @@ public void setProposedDeptName(String proposedDeptName) {
...
 	}
 
+	public String getUserAuthToken() {
+		return userAuthToken;
+	}
+
+	public void setUserAuthToken(String userAuthToken) {
+		this.userAuthToken = userAuthToken;
+	}
+
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/user/registration/model/UserRegistration.java:112]: Did we cross check the usage of userRegistration class ?
is there any side effect due to this change ?

Do test the user registration feature once this is deployed in PreProd

File: src/main/java/org/sunbird/user/service/UserUtilityServiceImpl.java [hunk 2]
```diff
@@ -589,9 +589,11 @@ public String createBulkUploadUser(UserRegistration userRegistration) {
 		requestBody.put(Constants.PHONE_VERIFIED, true);
 		requestBody.put(Constants.ROLES, Arrays.asList(Constants.PUBLIC));
 		request.put(Constants.REQUEST, requestBody);
+		Map<String, String> headerValues = ProjectUtil.getDefaultHeaders();
+		headerValues.put(Constants.X_AUTH_TOKEN, userRegistration.getUserAuthToken());
```
Comment (karthik-tarento) [inline src/main/java/org/sunbird/user/service/UserUtilityServiceImpl.java]: Need to add this header only if the value is available.
getDefaultHeaders adds content-type header -- is this added again ?

**Review by karthik-tarento — CHANGES_REQUESTED** [review]

File: src/main/java/org/sunbird/user/registration/model/UserRegistration.java (hunk 1 above)
Comment (Sahil-tarento) [reply src/main/java/org/sunbird/user/registration/model/UserRegistration.java:112]: @JsonIgnoreProperties(ignoreUnknown = true) is added on the entity class

File: src/main/java/org/sunbird/user/service/UserUtilityServiceImpl.java (hunk 2 above)
Comment (Sahil-tarento) [reply src/main/java/org/sunbird/user/service/UserUtilityServiceImpl.java]: added the empty check

**Review by karthik-tarento — APPROVED** [review]

**Pull request merged**